### PR TITLE
Make signal handler setup synchronous to avoid race conditions

### PR DESCRIPTION
**Pull request by niner**
On MoarVM setup of a signal handler is itself an asynchronous operation. Wait
for it to complete before returning as not only our spec tests but probably
every single user expects the signal handler to be in place at that point. This
was already true for the JVM and JS backends.

Fixes a race condition in t/spec/S17-procasync/kill.t

### CODE REVIEW

**Comment by MasterDuke17** [comment]
Relies on https://github.com/MoarVM/MoarVM/pull/1445

**Comment by niner** [comment]
I updated the commit to use a Semaphore instead of the Promise and await. With this, signal handler setup will behave synchronously even when done as part of react/whenever setup, making https://github.com/Raku/roast/commit/099d13e3b38fa48714b785076d5cf4d74f768fee obsolete.

File: src/core.c/signals.pm6 [hunk 1]
```diff
@@ -45,10 +45,24 @@ multi sub signal(*@signals, :$scheduler = $*SCHEDULER) {
             submethod BUILD(:$!scheduler, :$!signal) { }
 
             method tap(&emit, &, &, &tap) {
-                my $cancellation := nqp::signal($!scheduler.queue(:hint-time-sensitive),
+                my $queue := $!scheduler.queue(:hint-time-sensitive);
+#?if moar
+                my $setup-semaphore = Semaphore.new(0);
```
Comment (lizmat) [inline src/core.c/signals.pm6:50]: := perhaps?  Saves one Scalar and indicates immutability?

**Comment by niner** [comment]
> := perhaps?  Saves one Scalar and indicates immutability?

Right. Updated the commit.

**Comment by niner** [comment]
Rebased and merged manually (to get the NQP bump in)

**Pull request closed without merging**